packages/polling-controller/src/PollingController.ts

```diff
@@ -49,19 +59,26 @@ function PollingControllerMixin<TBase extends Constructor>(Base: TBase) {
      * Starts polling for a networkClientId
      *
      * @param networkClientId - The networkClientId to start polling for
+     * @param options - The options used to group the polling events
      * @returns void
      */
-    startPollingByNetworkClientId(networkClientId: NetworkClientId) {
+    startPollingByNetworkClientId(
+      networkClientId: NetworkClientId,
+      options: object = {},
+    ) {
       const innerPollToken = random();
-      if (this.#networkClientIdTokensMap.has(networkClientId)) {
-        const set = this.#networkClientIdTokensMap.get(networkClientId);
-        set?.add(innerPollToken);
+
+      const id = getKey(networkClientId, options);
+
+      const innerPollTokenSet = this.#networkClientIdTokensMap.get(id);
```

Comment:
Hmm.

It looks like the type of keys in this map is `NetworkClientId`, yet `id` is _not_ a network client ID, it's just a string. This means that the type of `this.#networkClientIdTokensMap` is misleading.

This _should_ have caused a type error... but did not. From a TypeScript perspective, `NetworkClientId` is equivalent to `string`, and so there is no problem here.

I don't feel like this is right. I feel like this reveals a limitation in how `NetworkClientId` is defined. If a network client ID were "special" — for instance, if it were prefixed with a special string such as `nc__` so that it could be defined as `nc__${string}` — then TypeScript would know that `id` is not a network client ID and we'd get the type error we're expecting.

So, first: What do we feel about prefixing network client IDs in this way?

Second: Since we couldn't make that type of change right now, in the meantime, what do we think about tweaking the type of `this.#networkClientIdTokensMap` to be more accurate? We could prefix the string that `getKey` returns in a similar way so that TypeScript sees it as "special" too. Then we could define a type such as

``` typescript
type PollTokenGroupId = `ptg__${string}`;
```

and then use that everywhere we are currently using `NetworkClientId` when we should be using something else.